packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,140 @@
+  stacked,
+  striped,
+  scrollTable,
+  scrollableCaption,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  useEffect(() => {
+    if (scrollTable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollable = scrollWidth > clientWidth;
+      setTabIndex(scrollable ? '0' : null);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const scrollable = scrollWidth > clientWidth;
+        setTabIndex(scrollable ? '0' : null);
+      }, 1000);
+
+      // Create event listener that calls handler function
+      window.addEventListener('resize', debouncedHandleResize);
+
+      // Remove event listener on cleanup
+      return () => {
+        window.removeEventListener('resize', debouncedHandleResize);
+      };
+    }
+  });
+
+  const classes = classNames(
+    'ds-c-table',
+    striped ? 'ds-c-table--striped' : null,
+    stacked ? `ds-c-table--stacked-${stacked}` : null,
+    className
+  );
+
+  const attributeScrollTable = scrollTable && {
+    className: 'ds-c-table__wrapper',
+    role: 'region',
+    'aria-labelledby': captionID.current,
+  };
+
+  const isTableCaptionComponents = (child) => {
+    return child && child.type === TableCaption;
+  };
+
+  const renderChildren = (captionId) => {
+    return React.Children.map(children, (child) => {
+      if (scrollTable && isTableCaptionComponents(child)) {
+        // Extend props on tables before rendering.
+        return React.cloneElement(child, {
+          id: captionId,
+        });
+      }
+
+      return child;
+    });
+  };
+
+  const renderScrollableCaption = () => {
+    const isScrollable = tabIndex === '0';
+    return scrollTable && isScrollable ? scrollableCaption : null;
+  };
+
+  return (
+    <>
+      {renderScrollableCaption()}
+      <div ref={container} tabIndex={tabIndex} {...attributeScrollTable}>
+        <table className={classes} role="table" {...attributeOptions}>
+          {renderChildren(captionID.current)}
+        </table>
+      </div>
+    </>
+  );
+};
+
+Table.defaultProps = {
+  className: '',
+  scrollableCaption: (
+    <div className="ds-u-margin-y--1">
+      <Alert>
+        <p className="ds-c-alert__text" aria-hidden="true">
+          Scroll using arrow keys to see more
+        </p>
+      </Alert>
+    </div>
+  ),
+};
+
+Table.propTypes = {
+  /**
+   * The table contents, usually `TableCaption`, `TableHead` and `TableBody`.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Additional classes to be added to the root table element.
+   */
+  className: PropTypes.string,
+  /**
+   * Responsive design breakpoint prefix to apply stack cells style at different viewpoint sizes.
+   */
+  stacked: PropTypes.oneOf(['sm', 'md', 'lg']),
```

Comment:
I'm wondering if we change this to `breakpoint` or `responsiveBreakpoint` ?

Reply:
or maybe even `responsiveTable`? or

Reply:
Actually on second thought, responsive might not be the best description considering horizontal scroll is also a way to make the table responsive. I think including `stack` in the naming does make sense. Maybe `stackBreakpoint`?